src/libaktualizr/uptane/fetcher.cc

```diff
@@ -22,24 +22,18 @@ static size_t DownloadHandler(char* contents, size_t size, size_t nmemb, void* u
   assert(userp);
   auto* ds = static_cast<Uptane::DownloadMetaStruct*>(userp);
   uint64_t downloaded = size * nmemb;
-  auto expected = static_cast<uint64_t>(ds->target.length());
+  uint64_t expected = ds->target.length();
   if ((ds->downloaded_length + downloaded) > expected) {
     return downloaded + 1;  // curl will abort if return unexpected size;
   }
 
   // incomplete writes will stop the download (written_size != nmemb*size)
   size_t written_size = ds->fhandle->wfeed(reinterpret_cast<uint8_t*>(contents), downloaded);
   ds->hasher().update(reinterpret_cast<const unsigned char*>(contents), written_size);
-  unsigned int calculated = 0;
-  if (loggerGetSeverity() <= boost::log::trivial::severity_level::trace) {
-    if (ds->downloaded_length > 0) {
-      std::cout << "\r";
-    }
-  }
   ds->downloaded_length += downloaded;
-  calculated = static_cast<unsigned int>((ds->downloaded_length * 100) / expected);
+  auto calculated = static_cast<unsigned int>((ds->downloaded_length * 100) / expected);
   if (loggerGetSeverity() <= boost::log::trivial::severity_level::trace) {
-    std::cout << "Downloading: " << calculated << "%";
+    std::cout << "\rDownloading: " << calculated << "%";
```

Comment:
It might be time to get rid of this `cout` business entirely. It's basically useless. If you are using trace debugging, there is way too much information to catch these, and the `\r` means if you go back looking for it, you only see `Downloading: 100%`. hmi-stub already prints out download progress; aktualizr probably should at debug or even info level in the same fashion as hmi-stub. On the other hand, I'm not sure why I didn't see the `Download status` bit that you added below.

Reply:
I agree, printing the events should be enough. Forgot `\n` in `Download status`, maybe that's the reason.